### ISSUE TITLE
Failed to list *coordination.Lease: unexpected EOF

### ISSUE THREAD

**Issue by alekc** [k3s-io/k3s#2335]
**Environmental Info:**
K3s Version: k3s version v1.18.9+k3s1 (630bebf9)

Node(s) CPU architecture, OS, and Version: 

Linux nas 4.15.18-30-pve #1 SMP PVE 4.15.18-58 (Fri, 12 Jun 2020 13:53:01 +0200) x86_64 GNU/Linux

Cluster Configuration:

2 masters in HA using external mysql server

**Describe the bug:**

This is my home cluster. After a power loss which affected both masters (Mysql is on another premises so it was not affected), the scheduling on cluster is kind of broken. 
Looking into logs I can see following 
```
Oct  2 23:34:03 nas k3s[13229]: E1002 23:34:03.109996   13229 cacher.go:428] unexpected ListAndWatch error: storage/cacher.go:/leases: Failed to list *coordination.Lease: unexpected EOF
Oct  2 23:34:03 nas k3s[13229]: E1002 23:34:03.148447   13229 cacher.go:428] unexpected ListAndWatch error: storage/cacher.go:/leases: Failed to list *coordination.Lease: unexpected EOF
Oct  2 23:34:03 nas k3s[13229]: E1002 23:34:03.306655   13229 cacher.go:428] unexpected ListAndWatch error: storage/cacher.go:/configmaps: Failed to list *core.ConfigMap: unexpected EOF
Oct  2 23:34:03 nas k3s[13229]: E1002 23:34:03.987163   13229 status.go:71] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"unexpected EOF"}
Oct  2 23:34:04 nas k3s[13229]: W1002 23:34:04.071522   13229 handler_proxy.go:102] no RequestInfo found in the context
Oct  2 23:34:04 nas k3s[13229]: E1002 23:34:04.072476   13229 controller.go:114] loading OpenAPI spec for "v1beta1.metrics.k8s.io" failed with: failed to retrieve openAPI spec, http error: ResponseCode: 503, Body: service unavailable
Oct  2 23:34:04 nas k3s[13229]: , Header: map[Content-Type:[text/plain; charset=utf-8] X-Content-Type-Options:[nosniff]]
Oct  2 23:34:04 nas k3s[13229]: I1002 23:34:04.073192   13229 controller.go:127] OpenAPI AggregationController: action for item v1beta1.metrics.k8s.io: Rate Limited Requeue.
Oct  2 23:34:04 nas k3s[13229]: E1002 23:34:04.214388   13229 cacher.go:428] unexpected ListAndWatch error: storage/cacher.go:/leases: Failed to list *coordination.Lease: unexpected EOF
Oct  2 23:34:04 nas k3s[13229]: E1002 23:34:04.467276   13229 cacher.go:428] unexpected ListAndWatch error: storage/cacher.go:/leases: Failed to list *coordination.Lease: unexpected EOF
Oct  2 23:34:04 nas k3s[13229]: I1002 23:34:04.857414   13229 trace.go:116] Trace[1435713186]: "List etcd3" key:/configmaps,resourceVersion:0,limit:500,continue: (started: 2020-10-02 23:34:03.80686837 +0100 BST m=+431.526402027) (total time: 1.050509371s):
Oct  2 23:34:04 nas k3s[13229]: Trace[1435713186]: [1.050509371s] [1.050509371s] END
Oct  2 23:34:04 nas k3s[13229]: E1002 23:34:04.858499   13229 status.go:71] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"unexpected EOF"}
Oct  2 23:34:04 nas k3s[13229]: I1002 23:34:04.859325   13229 trace.go:116] Trace[1483488246]: "List" url:/api/v1/configmaps,user-agent:operator/v0.0.0 (linux/amd64) kubernetes/$Format,client:192.168.0.29 (started: 2020-10-02 23:34:03.806821588 +0100 BST m=+431.526355250) (total time: 1.052467125s):
Oct  2 23:34:04 nas k3s[13229]: Trace[1483488246]: [1.052467125s] [1.052428814s] END
```

investigating a bit further I can see that I indeed cannot obtain a list of leases in kube-system namespace
```
╰─ kubectl get lease                                                                                                                   ─╯
Error from server: unexpected EOF
```

**Comment by brandond** [comment]
Can you try the following:
1. Stop k3s on both servers; wait at least 30 seconds
1. Start k3s on one server; wait at least 60 seconds
1. Start k3s on the other server

If you continue to receive errors, please gather logs from both servers from startup onwards.Just having the tail of the logs makes it difficult to determine what the initial error was.

**Comment by alekc** [comment]
Sure thing 
I also tried to reboot both servers. 

In attached log you will find a startup sequence for one of the nodes, and a panic printout I oserved on it (outside of the launch sequence). 

[panic.txt](https://github.com/rancher/k3s/files/5321316/panic.txt)

[k3s.log](https://github.com/rancher/k3s/files/5321314/k3s.log)

By the look of it, something has corrupted a core config on the db, which prevents a scheduler from running. 
Probably I am going to make a backup of the current db to save that state and drop kine table to rebuild the cluster.
